### ISSUE TITLE
Report if comparison is done via "Equals(true)" or "Equals(false)"

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#2006]
We should issue a warning and suggest to simplify in case the code uses a comparison via `.Equals(true)` or `.Equals(false)`, followed by comparison with `== true` or `is true`.

Example:

Following code should trigger a warning:
```c#
public void DoSomething(object o)
{
    if (o?.Something?.Else?.Stuff?.Equals(true) == true)
    {
        // do stuff ...
    }
}
```

It should be fixed into:
```c#
public void DoSomething(object o)
{
    if (o?.Something?.Else?.Stuff is true)
    {
        // do stuff ...
    }
}
```

Same is valid for:
```c#
public void DoSomething(object o)
{
    if (o?.Something?.Else?.Stuff?.Equals(false) == true)
    {
        // do stuff ...
    }
}
```

which should be fixed into:
```c#
public void DoSomething(object o)
{
    if (o?.Something?.Else?.Stuff is false)
    {
        // do stuff ...
    }
}
```
